go.mod

```diff
@@ -32,6 +32,8 @@ require (
 )
 
 require (
+	github.com/kyma-project/template-operator v0.0.0-20230925081606-4ae25406be03
```

Comment:
I use it in e2e testing to keep the code as simple as possible, is there a way I can exclude this fomr go.mod since it is just used in test files?

Reply:
no good way to exclude it, otherwise you need to design this test as a dedicate go module, and then you can create new go.mod for this testing module, but it's not a practical way.

Anyway, template-operator is not needed, the only benefit you can get is to directly reference some API code, like the typed Sample CR, but you can also created it with `Unstructured{}`.

Instead, introducing it brings lots of dependencies problems, for example, both of them are operator, we may introduce different controller-runtime versions, think about later controller-runtime have breaking changes, you may blocked LM to be adapted smoothly because the template-operator still using old version.

We need to really be care to avoid introducing unnecessary dependencies.

Reply:
But additionally, if we can refactor template operator API as a dedicated go module, then we can add it with introduce fewer dependencies. That's also one [idea](https://github.com/orgs/kyma-project/projects/30/views/13?pane=issue&itemId=38579080) I'm plan to do it for LM api.